u8timeseries/timeseries.py

```diff
@@ -135,49 +216,89 @@ def _slice_not_none(s: Optional[pd.Series]) -> Optional[pd.Series]:
 
     def slice_n_points_after(self, start_ts: pd.Timestamp, n: int) -> 'TimeSeries':
         """
-        Returns a new time series, starting later than [start_ts] (inclusive) and having (at most) [n] points
-        :param start_ts:
-        :param n:
-        :return:
+        Returns a new TimeSeries, starting later than [start_ts] and having (at most) [n] points.
+
+        The timestamp may not be in the time series. If it is, it will be included in the new TimeSeries.
+
+        :param start_ts: The timestamp that indicates the splitting time.
+        :param n: The maximal length of the new TimeSeries.
+        :return: A new TimeSeries, with length at most [n] and indices greater or equal than [start_ts].
         """
-        end_ts: pd.Timestamp = start_ts + (n-1) * self.freq()  # (n-1) because slice() is inclusive on both sides
+
+        assert n >= 0, 'n should be a positive integer.'
+
+        start_ts = self.time_index()[self.time_index() >= start_ts][0]  # closest index after start_ts (new start_ts)
+
+        end_ts: pd.Timestamp = start_ts + (n - 1) * self.freq()  # (n-1) because slice() is inclusive on both sides
         return self.slice(start_ts, end_ts)
 
     def slice_n_points_before(self, end_ts: pd.Timestamp, n: int) -> 'TimeSeries':
         """
-        Returns a new time series, ending before [end_ts] (inclusive) and having (at most) [n] points
-        :param end_ts:
-        :param n:
-        :return:
+        Returns a new TimeSeries, ending before [end_ts] and having (at most) [n] points.
+
+        The timestamp may not be in the TimeSeries. If it is, it will be included in the new TimeSeries.
+
+        :param end_ts: The timestamp that indicates the splitting time.
+        :param n: The maximal length of the new time series.
+        :return: A new TimeSeries, with length at most [n] and indices smaller or equal than [end_ts].
         """
+
+        assert n >= 0, 'n should be a positive integer.'
+
+        end_ts = self.time_index()[self.time_index() <= end_ts][-1]
+
         start_ts: pd.Timestamp = end_ts - (n - 1) * self.freq()  # (n-1) because slice() is inclusive on both sides
         return self.slice(start_ts, end_ts)
 
     def intersect(self, other: 'TimeSeries') -> 'TimeSeries':
         """
-        Returns a slice containing the intersection of this TimeSeries and the one provided in argument
-        :param other:
-        :return:
+        Returns a slice containing the intersection of this TimeSeries and the one provided in argument.
+
+        :param other: A second TimeSeries.
+        :return: A new TimeSeries, with values of this TimeSeries and indices the intersection of both
+                TimeSeries' indices.
         """
-        return self.slice(other.start_time(), other.end_time())
```

Comment:
Yes, otherwise the returned TimeSeries is a restriction and not an intersection (this mainly affects cases of TimeSeries with different frequencies) A check for size has been added as a precaution.

Reply:
cool, makes sense indeed :)